mantidproject/mantid
# Add warnings of non existent periods

**Description of work.**

This adds warnings to the Muon Analysis GUI when one or more runs do not contain all the relevant periods to calculate a group fully. If some runs are missing periods the rows are colored in yellow and if all runs are missing periods they are colored in red. The tooltips are also updated to reflect this.

The names of groups calculated have also been updated to remove the periods information as this is now contained in the group.



**To test:**
You will need access to the ISIS data archive:
  * Load in a run with single period data and one with several periods. HIFI84447 and HIFI44447 should suffice for this.
  * Check that warnings are shown for all groups containing a period that is not 1.
  * Check that is you switch to just 44447 the yellow rows become red as they are now invalid for all groups.
  * Check that if you switch to just 84447 all the rows clear.



Fixes #29121 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
